packages/serverpod_serialization/lib/src/auth_encoding.dart

```diff
@@ -0,0 +1,62 @@
+/// Serverpod's default authentication key encoding and decoding.
+library serverpod_serialization.auth_encoding;
+
+import 'dart:convert';
+
+/// The name of the default Serverpod scheme for HTTP "authorization" headers.
+/// Note, the scheme name is case-insensitive and should be compared in a case-insensitive manner.
+const serverpodDefaultAuthSchemeName = 'Basic';
+
+/// Regexp for a non-empty string adhering to the RFC 4648 base64 encoding alphabet.
+const _base64RegExpStr = r'[a-zA-Z0-9+/]*=*';
+
+/// The regular expression for the format of the authentication scheme.
+const _authSchemeRegExpStr = r'[a-zA-Z0-9_-]+';
+
+/// The regular expression for the format of the value in an HTTP header.
+/// This also applies to the authentication data payload / key.
+/// It consists of:
+/// - \t (tab character)
+/// - \x20-\x7E (printable ASCII characters, including space)
+/// - \x80-\xFF (extended ASCII characters)
+const _httpHeaderValueRegExpStr = r'[\t\x20-\x7E\x80-\xFF]*';
+
+/// The regular expression for the format of the HTTP "authorization" header value,
+/// including starting with an authentication scheme name.
+final _authValueRegExp =
+    RegExp('^$_authSchemeRegExpStr $_httpHeaderValueRegExpStr\$');
```

Comment:
Question: Just to double check, why is the dollar sign at the end escaped? I would have expected it to to not be escaped to signify end of string match, but perhaps I'm misunderstanding this?

Reply:
It's a dart string escape to prevent the dart string interpolation; yes it represents end-of-string match.

Reply:
Ohh alright, that's strange given that nothing follows the dollar sign? 🤔 Anyway, then I see, thanks!